lm_eval/api/task.py

```diff
@@ -373,13 +373,21 @@ def build_all_requests(
         world_size=None,
         cache_requests=False,
         rewrite_requests_cache=False,
+        system_instruction=None,
+        apply_chat_template=False,
+        fewshot_as_multiturn=False,
+        lm=None,
     ) -> None:
         """Build a set of Instances for a task, and store them in task.instances"""
 
         # used with caching
         og_limit = limit
 
         cache_key = f"requests-{self._config.task}-{self.config.num_fewshot}shot-rank{rank}-world_size{world_size}"
+        cache_key += "-chat_template" if apply_chat_template else ""
```

Comment:
We'll need to also add (the hash of) the system prompt to cache key since that'll affect the requests that are created.